File: examples/plugin-archives/demo-linked-enum-table/python/provider.py
def execute(params, upstream, context):
    import pandas as pd
    rows = max(2, int(params.get("rows", 6)))
    profile = str(params.get("profile", "linear"))
    series = str(params.get("series", "value"))
    scale = float(params.get("scale", 2))
    power = max(2, int(params.get("power", 2)))
    values = []
    for index in range(rows):
        if profile == "polynomial":
            exponent = 3 if series == "cube" else 2 if series == "square" else power
            values.append(index ** exponent)
        else:
            values.append(index * (2 if series == "double" else scale))
    return pd.DataFrame({"index": list(range(rows)), "selected": values})

File: examples/plugin-archives/demo-linked-enum-table/js/provider.js
function execute(params, upstream, context, api) {
  const rows = Math.max(2, Math.trunc(Number(params.rows ?? 6)));
  const profile = String(params.profile ?? "linear");
  const series = String(params.series ?? "value");
  const scale = Number(params.scale ?? 2);
  const power = Math.max(2, Math.trunc(Number(params.power ?? 2)));
  const records = Array.from({ length: rows }, (_, index) => {
    let selected;
    if (profile === "polynomial") selected = Math.pow(index, series === "cube" ? 3 : series === "square" ? 2 : power);
    else selected = index * (series === "double" ? 2 : scale);
    return { index, selected };
  });
  return api.Table.fromRecords(records);
}
